engine/communication/src/message.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Request {
+    Ping,
+    Publish(MailboxRecord),
+    Message(String),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MailboxRecord {
+    key: String,
+    value: String,
+    expires_sec: u64,
+}
+
+impl MailboxRecord {
+    pub fn new(key: String, value: String, expires_sec: u64) -> Self {
```

Comment:
quoting Lucas:

> Exposing `u64` values to JavaScript will result in the use of [BigUint64Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigUint64Array) in the generated JS wrapper code, and it isn't supported on Safari. So it's recommended to use `f64` instead, see [this PR on iota.rs](https://github.com/iotaledger/iota.rs/pull/94).

Reply:
since the u64 is required by kademlia and for now this wont be exposed to the outside of the application, I'd ignore this for now but keep it in mind for later changes